File: src/main/java/com/saybetter/domain/member/dao/repository/MemberReadRepository.java
package com.saybetter.domain.member.dao.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.saybetter.domain.member.domain.Member;
import com.saybetter.global.common.constant.Provider;

public interface MemberReadRepository extends JpaRepository<Member, Long> {

	Optional<Member> findByProviderAndLoginId(Provider provider, String loginId);

	Optional<Member> findByEmail(String email);
}

File: storage/storage-mysql-kt/src/main/kotlin/io/say/better/storage/mysql/domains/account/repository/MemberReadRepository.kt
package io.say.better.storage.mysql.domains.account.repository

import io.say.better.core.common.constant.Provider
import io.say.better.storage.mysql.domains.account.entity.Member
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface MemberReadRepository : JpaRepository<Member, Long> {
    fun findByProviderAndLoginId(
        provider: Provider,
        loginId: String,
    ): Optional<Member>

    fun findByEmail(email: String): Optional<Member>
}
